Server/Components/Pages/GamePage.razor

```diff
@@ -54,10 +54,22 @@
 
 @code {
 	private List<Item> equipment = new();
-	private PopupContainer popup;
+	private List<Item> inventory = new();
+
+    Dictionary<string, object> equipmentPopupParameters = new();
+	
+    private PopupContainer popup;
 
 	protected override Task OnInitializedAsync()
 	{
-		return base.OnInitializedAsync();
+	    equipmentPopupParameters.Add("inventory", inventory);
```

Comment:
hmmmm, i don't really like it that much either...

the only other way i think of is using a collection initializer, like so:
```csharp
equipmentPopupParameters = new()
{
    {"inventory", inventory},
    {"equipment", equipment},
    {"OnEquipmentChanged", EventCallback.Factory.Create(this, HandleEquipmentChanged)}
};
```

not that it's any prettier, so i'll leave it up to your own discretion to decide on which style you want to go with.

in the future, i'd love some kind of a fluent builder for constructing parameter objects like these

Reply:
I agree this isn't ideal, but I like your idea better - it is easier to see what goes into certain parameters (when we will have multiple dictionaries like this one).